### ISSUE TITLE
EBNF for <identifier>

### ISSUE THREAD

**Issue by kaby76** [dart-lang/language#1331]
I think the EBNF for <identifier> in the [spec](https://github.com/dart-lang/language/blob/master/specification/dartLangSpec.tex) is wrong. It is stated as follows (not in Latex, but as it appears in the .pdf):

    <identifier> ::= <IDENTIFIER> .

Notice the "." (DOT) after <IDENTIFIER>. This must a typo because <IDENTIFIER> is defined:

    <IDENTIFIER> ::= <IDENTIFIER_START> <IDENTIFIER_PART>*

I know of no other language which defines an identifier with a trailing ".". This makes no sense. No other EBNF rule ends in a period, but it is here for <identifier>.

Can someone please comment?

(I am scraping the grammar from the spec. Afterward this, I will scrape the grammar from the implementation itself, because I do not trust what anyone ever types.)

**Comment by lrhn** [comment]
@eernstg Is this a leftover from a previous way of specifying the grammar?

**Comment by kaby76** [comment]
I'm looking at line 14156 of [https://raw.githubusercontent.com/dart-lang/language/master/specification/dartLangSpec.tex](https://raw.githubusercontent.com/dart-lang/language/master/specification/dartLangSpec.tex). For whatever reason, Github.com does not display the Latex file correctly, showing only 13564 lines. I know because I cloned this and I'm working on "master" aka "main" or whatever they want to call the main branch nowadays or past. I would provide a link to the exact line but Github is not working.

**Comment by eernstg** [comment]
True, we should remove that `.`, and it is already on my todo list of small fixes.

@kaby76, if you are scraping the grammar from the spec in order to use it for actual parsing you may also wish to take a look at [Dart.g](https://github.com/dart-lang/sdk/blob/master/tools/spec_parser/Dart.g), which is an ANTLR v4 specification that is kept in alignment with the grammar in the language specification, but also models things like the special treatment of `yield` and `await` when in a `sync*`, `async`, or `async*` function body.

The problem with github only showing about 14K lines from `dartLangSpec.tex` is known, it has been like that for quite a while. A reload may solve the problem, but not reliably. There should not be a conflict with the [github limits](https://docs.github.com/en/free-pro-team@latest/github/creating-cloning-and-archiving-repositories/limits-for-viewing-content-and-diffs-in-a-repository#text-limits). I've reported the issue recently, but not yet received a response.

**Comment by kaby76** [comment]
> take a look at [Dart.g](https://github.com/dart-lang/sdk/blob/master/tools/spec_parser/Dart.g)

Excelllent. I'll apply that to the grammar diff program I am writing.
